### PR TITLE
Don't break MMRests on invisible elements

### PR DESCRIPTION
**Pull request by miiizen**
Resolves: #24812 
Resolves: https://github.com/musescore/MuseScore/issues/15863

If an item is invisible, don't break MMRests and don't lay the item out (in the case of spanners)

### CODE REVIEW

File: src/engraving/dom/spanner.cpp [hunk 1]
```diff
@@ -1153,6 +1153,45 @@ Measure* Spanner::endMeasure() const
...
+    if (!m_startElement) {
+        return nullptr;
+    }
+
+    if (m_startElement->isChordRest()) {
+        return toChordRest(m_startElement)->measure();
+    } else if (m_startElement->isSegment()) {
+        return toSegment(m_startElement)->measure();
+    } else if (m_startElement->isMeasure()) {
+        toMeasure(m_startElement);
```
Comment (cbjeukendrup) [inline src/engraving/dom/spanner.cpp:1167]:
```suggestion
        return toMeasure(m_startElement);
```

File: src/engraving/dom/spanner.cpp [hunk 2]
```diff
@@ -1153,6 +1153,45 @@ Measure* Spanner::endMeasure() const
...
+{
+    if (!m_endElement) {
+        return nullptr;
+    }
+    if (m_endElement->isChordRest()) {
+        return toChordRest(m_endElement)->measure();
+    } else if (m_endElement->isSegment()) {
+        return toSegment(m_endElement)->measure();
+    } else if (m_endElement->isMeasure()) {
+        toMeasure(m_endElement);
```
Comment (cbjeukendrup) [inline src/engraving/dom/spanner.cpp:1186]:
```suggestion
        return toMeasure(m_endElement);
```

File: src/engraving/dom/engravingitem.cpp [hunk 3]
```diff
@@ -1462,6 +1462,13 @@ PropertyPropagation EngravingItem::propertyPropagation(const EngravingItem* dest
     const Score* sourceScore = score();
     const Score* destinationScore = destinationItem->score();
     const bool isTextProperty = propertyGroup(propertyId) == PropertyGroup::TEXT;
+    const Staff* sourceStaff = staff();
+    const Staff* destinationStaff = destinationItem->staff();
+
+    // Properties must be propagated to items cloned for MMRests
+    if (sourceScore == destinationScore && sourceStaff == destinationStaff) {
```
Comment (miiizen) [inline src/engraving/dom/engravingitem.cpp:1469]: Currently, properties aren't propagated to items in the same score.  I think they definitely should be for clones inside/outside MMRests - see below.

https://github.com/user-attachments/assets/751db5fd-c5ab-494d-a2ac-a0b8d680c253

It doesn't seem the most robust way of detecting this specific situation, but I don't think we ever have clones on the same score & staff apart from with MMRests.

**Review by oktophonie — APPROVED** [review]

File: src/engraving/dom/spanner.cpp [hunk 4]
```diff
@@ -1153,6 +1153,45 @@ Measure* Spanner::endMeasure() const
     return toMeasure(m_endElement);
 }
 
+Measure* Spanner::findStartMeasure() const
+{
+    if (!m_startElement) {
+        return nullptr;
+    }
+
+    if (m_startElement->isChordRest()) {
```
Comment (mike-spa) [inline src/engraving/dom/spanner.cpp:1162]: If I'm not mistaken, I think you could replace this entire method with simply 
`m_startElement->findAncestor(ElementType::MEASURE);`
Same for the end element.

**Review by mike-spa — APPROVED** [review]

**Pull request merged**